nl/seg6_linux.go

```diff
@@ -0,0 +1,121 @@
+package nl
+
+import (
+	"errors"
+	"fmt"
+	"net"
+)
+
+type Ipv6SrHdr struct {
+	nexthdr       uint8
+	hdrlen        uint8
+	typ           uint8
+	segments_left uint8
+	first_segment uint8
+	flags         uint8
+	reserved      uint16
+
+	Segments []net.IP
+}
+
+func (s1 *Ipv6SrHdr) Equal(s2 Ipv6SrHdr) bool {
+	if len(s1.Segments) != len(s2.Segments) {
+		return false
+	}
+	if len(s1.Segments) != 0 {
```

Comment:
Minor comment: This `if` check is not really needed, if no segments are there, the `for` loop will be a no op.

Reply:
Thanks for the review. I will remove this `if`.